common/libraries/clients/postgresql/src/test/java/org/eclipse/dataspaceconnector/clients/postgresql/PooledPostgresqlClientTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql;
+
+import org.eclipse.dataspaceconnector.clients.postgresql.connection.ConnectionFactory;
+import org.eclipse.dataspaceconnector.clients.postgresql.connection.ConnectionFactoryConfig;
+import org.eclipse.dataspaceconnector.clients.postgresql.connection.ConnectionFactoryImpl;
+import org.eclipse.dataspaceconnector.clients.postgresql.connection.pool.ConnectionPool;
+import org.eclipse.dataspaceconnector.clients.postgresql.connection.pool.commons.CommonsConnectionPool;
+import org.eclipse.dataspaceconnector.clients.postgresql.connection.pool.commons.CommonsConnectionPoolConfig;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+class PooledPostgresqlClientTest extends AbstractPostgresqlClientTest {
+    private PostgresqlClient postgresqlClient;
```

Comment:
the client could stay in the abstraction, being instantiated by an abstract method (that would do the same things of the `setUp`, this will avoid the duplication of the teardown and the need for that `getPostgresqlClient`.

Reply:
Once the `SimplePostgresClient` is made package private this will be refactored - so the abstract test class will go away anyway :)